pkg/controller/iam/iamclient/helpers.go

```diff
@@ -54,6 +56,26 @@ func ResolveMemberIdentity(ctx context.Context, member v1beta1.Member,
 		return string(member), nil
 	}
 
+	if err := memberFrom.Validate(); err != nil {
+		return id, err
+	}
+
+	if id, err := tryResolveTFMemberReference(ctx, memberFrom, namespace, tfIAMClient); err != nil {
```

Comment:
I like the name, though it is a little bit misleading. For example, the tryResolveTFMemberRefer contains sqlinstance  which could be a direct resource, so it's not "successfully resolved member reference from TF-based resources".

Reply:
Hmm... I named the function this way because all 4 resources referenced within this function depend on the TF-based service mapping to determine the service account. As an example, for SQLInstance it uses this configuration.
https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/3530c83a5e0d331640ec2160675d80336fad9c53/config/servicemappings/sql.yaml#L56-L58

/cc @jasonvigil When SQLInstance is switched to a direct resource, do we still depend on service mapping when the SQLInstance is referenced by an IAM resource? e.g.
```
apiVersion: iam.cnrm.cloud.google.com/v1beta1
kind: IAMPolicyMember
metadata:
  name: iampolicymember-sample-memberref
spec:
  memberFrom:
    sqlInstanceRef:
      name: xxx
```